[paklongchiu] iP

paklongchiu opened the pull request.

Commit:
Level-0

Commit:
Revert "Level-0"

This reverts commit a8435c7934e0ebcddd39ac378c7d67a3d9afd844.

Commit:
Reapply "Level-0"

This reverts commit 1f4bfd9d9b23ffc7f9ab2a28ebabd552d2c029f2.

Commit:
Enable ability to echo user's input, exit when it is bye

Commit:
Enable the ability to store user's inputs and display them when requested

Commit:
Add functions to mark and unmark tasks, while adopting a Task class to do so

Commit:
Optimise the way of getting input and check if the code aligns with coding standard

Ashertan256 commented on src/main/java/Elyk.java:
```diff
@@ -0,0 +1,79 @@
+    public static void main(String[] args) {
+        greet();
+
+        while (true) {
```
Instead of while(true), maybe you can consider using this instead:

do {
  // code to run
}
while (!input.equal("bye"));

Ashertan256 commented on src/main/java/Elyk.java:
```diff
@@ -0,0 +1,79 @@
+    public static void markTaskNotDone(int taskNum) {
+        taskList[taskNum - 1].markAsNotDone();
+        System.out.println("OK, I've marked this task as not done yet:");
+        System.out.println("  [ ] " + taskList[taskNum - 1].description);
```
Would using taskList[i].getStatusIcon (as you used below) be more consistent?

Ashertan256 commented on src/main/java/Elyk.java:
```diff
@@ -0,0 +1,79 @@
+
+    public static void updateInput() {
+        input = command.nextLine();
+        if (input.matches(".*\\d$")) {
```
A comment could be added to describe the regex pattern to be clearer

Ashertan256 commented on src/main/java/Elyk.java:
```diff
@@ -0,0 +1,79 @@
+        if (input.matches(".*\\d$")) {
+            String[] words = input.split(" ");
+            input = words[0];
+            taskNum = Integer.parseInt(words[1]);
```
This could potentially have issues if the second word is not a number. Attempting to parseInt a String may crash the program/cause an exception

Ashertan256 commented on src/main/java/Elyk.java:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    public static void markTaskNotDone(int taskNum) {
+        taskList[taskNum - 1].markAsNotDone();
```
This seems like a redundant method, is it possible to call a single method without chaining multiple methods that do the same thing?

Ashertan256 reviewed (commented):
Overall, the structure and flow of the code seems a little redundant at certain parts. But the format looks to be consistent with the course!

Commit:
Using inheritance to support tracking 3 types of tasks: Todo, Deadline and Event

Commit:
Added some exceptions for dealing with errors,
like input task type is not supported or there is missing information from the input

thomasjlalba commented on src/main/java/Elyk.java:
```diff
@@ -0,0 +1,110 @@
+            int toPos = input.indexOf("/to");
+            to = input.substring(toPos + 4);
+            from = input.substring(fromPos + 6, toPos - 1);
+            description = input.substring(6, fromPos - 1);
```
Maybe you can[ refactor the magic literals](https://nus-cs2113-ay2425s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers) throughout your code to make it easier to understand

thomasjlalba commented on src/main/java/Elyk.java:
```diff
@@ -0,0 +1,110 @@
+            from = input.substring(fromPos + 6, toPos - 1);
+            description = input.substring(6, fromPos - 1);
+            input = "event";
+        }
```
Maybe you can [include the final else statement](https://nus-cs2113-ay2425s2.github.io/website/se-book-adapted/chapters/codeQuality.html#use-the-default-branch) for error detection

thomasjlalba commented on src/main/java/Elyk.java:
```diff
@@ -0,0 +1,110 @@
+    public static void printTask() {
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < taskCounter; i++) {
+            System.out.println((i+1) + "." + taskList[i]);
```
Maybe you can standardise the use of spacing within your `i+1` to make it more consistent (`i + 1` instead)

thomasjlalba commented on src/main/java/Elyk.java:
```diff
@@ -0,0 +1,110 @@
+                    markTaskNotDone(taskNum);
+                    break;
+                case "todo":
+                    taskList[taskCounter++] = new Todo(description);
```
It might be more readable by separating the increment to the next line instead of combining it in the same line

thomasjlalba commented on src/main/java/Elyk.java:
```diff
@@ -0,0 +1,110 @@
+import java.util.Scanner;
+
+public class Elyk {
+    public static int taskCounter = 0;
+    public static int taskNum = 0;
```
The naming of these variables might suggest that they are the same. Maybe you can consider renaming them to make them clearer and more distinct.

thomasjlalba commented on src/main/java/Elyk.java:
```diff
@@ -0,0 +1,110 @@
+    public static Scanner command = new Scanner(System.in);
+
+    public static void main(String[] args) {
+        greet();
```
Good job on the abstraction of code to make it neater!

thomasjlalba reviewed (commented):
Good job thus far! Just some trivial suggestions to make it more readable and understandable for others.

Commit:
Using ArrayList<Task> instead of an array to store tasks and turned Task into an abstract class

Commit:
Enabled deleting tasks from the list by task index

Commit:
Replaced taskCounter with taskList.size() and utilised taskList.getLast()

Commit:
Created a new class Storage to deal with loading and saving tasks in a file

Commit:
Made amendments in parseTaskFromString() so that there is no reassignment of local variable

Commit:
Merge branch 'branch-Level-6'

Commit:
Merge branch 'branch-Level-7'

Commit:
Added loadTasks() and saveTasks() into main() to complete the task saving function

Commit:
Fixed the issue of displaying wrong no. of tasks in deleteTask()

Commit:
Implemented a new class TaskList to contain the task list in Elyk and created Ui and Parser class

Modified code in Elyk and Storage to use TaskList instead

Commit:
Implemented class Ui but yet to incorporate its methods in other classes

Commit:
Completed implementing all new classes for OOP and modified the logic in Elyk to work with them

Commit:
Added "find" command case and method to extract keyword in Parser class

Commit:
Updated Ui and TaskList classes to find and display matching tasks according to keyword

Commit:
Updated main logic in Elyk to support finding tasks by a keyword

Commit:
Added JavaDoc comments to the Ui class and all of its methods

Commit:
Added JavaDoc comments to 3 more classes and all of their non-private methods

Commit:
Completed the remaining JavaDoc comments for all classes and methods

paklongchiu committed:
Merge pull request #1 from paklongchiu/branch-Level-9

Branch level 9

Commit:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/Ui.java

paklongchiu committed:
Merge pull request #2 from paklongchiu/branch-A-JavaDoc

Branch a java doc

Commit:
Explained the 9 main features of Elyk in UG
